code/modules/mob/living/simple_animal/hostile/regalrat.dm

```diff
@@ -94,27 +94,25 @@
 	return ..()
 
 /mob/living/simple_animal/hostile/regalrat/CanAttack(atom/the_target)
-	if(istype(the_target,/mob/living/simple_animal))
-		var/mob/living/A = the_target
-		if(isregalrat(the_target) && A.stat == CONSCIOUS)
+	if(isliving(the_target))
+		var/mob/living/living_target = the_target
+		if(living_target.faction_check_mob(src, TRUE))
+			return FALSE
+		else
 			return TRUE
```

Comment:
```suggestion
		return living_target.faction_check_mob(src, TRUE)
```

Reply:
Do you mean to invert? (Also TRUE should have a named parameter there)

Reply:
```suggestion
		return !living_target.faction_check_mob(src, wat = TRUE)
```